math/mathcore/inc/TMath.h

```diff
@@ -1002,6 +1006,50 @@ Iterator TMath::LocMin(Iterator first, Iterator last) {
    return std::min_element(first, last);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Returns the one-dimensional gradient of a vector with length n. The default step size h is 1.
+
+template <typename T>
+T *TMath::Gradient(Long64_t n, T *f, T h)
+{
+   Long64_t i = 1;
+   T *result = new T[n + 1];
```

Comment:
```suggestion
   T *result = new T[n - 1];
```

Reply:
Thank you for providing suggestions. I noticed that if apply this suggestion and try to compile, I get this error

`malloc(): corrupted top size`

Also valgrind gives the following error.

<details>
==785512== Invalid write of size 8
==785512==    at 0x10952E: double* Gradient<double>(long long, double*, double) (f.h:18)
==785512==    by 0x109387: main (main.cpp:15)
==785512==  Address 0x4dd9cc8 is 0 bytes after a block of size 72 alloc'd
==785512==    at 0x484A2F3: operator new[](unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==785512==    by 0x10943A: double* Gradient<double>(long long, double*, double) (f.h:7)
==785512==    by 0x109387: main (main.cpp:15)
==785512== 

==785512== Invalid write of size 8
==785512==    at 0x10975E: double* Laplacian<double>(long long, double*, double) (f.h:37)
==785512==    by 0x1093AA: main (main.cpp:16)
==785512==  Address 0x4dd9d58 is 0 bytes after a block of size 72 alloc'd
==785512==    at 0x484A2F3: operator new[](unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==785512==    by 0x10957F: double* Laplacian<double>(long long, double*, double) (f.h:26)
==785512==    by 0x1093AA: main (main.cpp:16)
</details>

With `T *result = new T[n + 1];` there are no errors.